From 2717a24cdd4cc81026181116d490f979a778fa51 Mon Sep 17 00:00:00 2001
From: Craig Kewley <craigx.kewley@intel.com>
Date: Fri, 6 Apr 2018 11:19:48 +0100
Subject: [PATCH 090/102] ASoC: Intel: Boards: Add separate BE DAI for each
 Dirana pipe

This resolves an issue were the wrong HW params would be used. More details
can be found in commit "ASoC: Intel: Skylake: Add secondary backend dais"

Signed-off-by: Craig Kewley <craigx.kewley@intel.com>
---
 sound/soc/intel/boards/bxt_gpmrb.c | 46 ++++++++++++++++++++++++++++++++------
 1 file changed, 39 insertions(+), 7 deletions(-)

diff --git a/sound/soc/intel/boards/bxt_gpmrb.c b/sound/soc/intel/boards/bxt_gpmrb.c
index c8b6dca..39fcfcd 100644
--- a/sound/soc/intel/boards/bxt_gpmrb.c
+++ b/sound/soc/intel/boards/bxt_gpmrb.c
@@ -117,14 +117,14 @@ static int bt_modem_sample_rate_put(struct snd_kcontrol *kcontrol,
 	{ "Speaker", NULL, "ssp4 Tx"},
 	{ "ssp4 Tx", NULL, "codec0_out"},
 
-	{ "tuner_in", NULL, "ssp2 Rx"},
-	{ "ssp2 Rx", NULL, "DiranaTunerCp"},
+	{ "tuner_in", NULL, "ssp2-b Rx"},
+	{ "ssp2-b Rx", NULL, "DiranaTunerCp"},
 
-	{ "aux_in", NULL, "ssp2 Rx"},
-	{ "ssp2 Rx", NULL, "DiranaAuxCp"},
+	{ "aux_in", NULL, "ssp2-c Rx"},
+	{ "ssp2-c Rx", NULL, "DiranaAuxCp"},
 
-	{ "mic_in", NULL, "ssp2 Rx"},
-	{ "ssp2 Rx", NULL, "DiranaMicCp"},
+	{ "mic_in", NULL, "ssp2-d Rx"},
+	{ "ssp2-d Rx", NULL, "DiranaMicCp"},
 
 	{ "DiranaPb", NULL, "ssp2 Tx"},
 	{ "ssp2 Tx", NULL, "dirana_out"},
@@ -511,11 +511,43 @@ static int broxton_gpmrb_tdf8532_startup(struct snd_pcm_substream *substream)
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
 		.platform_name = "0000:00:0e.0",
-		.dpcm_capture = 1,
 		.dpcm_playback = 1,
 		.no_pcm = 1,
 	},
 	{
+		/* SSP2-B - Dirana */
+		.name = "SSP2-B-Codec",
+		.be_id = 1,
+		.cpu_dai_name = "SSP2-B Pin",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.platform_name = "0000:00:0e.0",
+		.dpcm_capture = 1,
+		.no_pcm = 1,
+	},
+	{
+		/* SSP2-C - Dirana */
+		.name = "SSP2-C-Codec",
+		.be_id = 1,
+		.cpu_dai_name = "SSP2-C Pin",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.platform_name = "0000:00:0e.0",
+		.dpcm_capture = 1,
+		.no_pcm = 1,
+	},
+	{
+		/* SSP2-D - Dirana */
+		.name = "SSP2-D-Codec",
+		.be_id = 1,
+		.cpu_dai_name = "SSP2-D Pin",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.platform_name = "0000:00:0e.0",
+		.dpcm_capture = 1,
+		.no_pcm = 1,
+	},
+	{
 		/* SSP3 - Modem */
 		.name = "SSP3-Codec",
 		.be_id = 1,
-- 
1.9.1

